docs/dev/network-protocol/establishing_connection.md

```diff
@@ -0,0 +1,161 @@
+# Establishing a connection
+
+This section describes the communication between the server and the client
+during connection establishing, note that only the TCP connection needs
+to be established for the client to be connected. After this the client
+will be visible to the other clients on the server and able to send other
+types of messages.
+
+## Connect
+
+As the basis for the synchronization procedure the client has to first
+establish the TCP connection to the server and do a common TLSv1 handshake.
+To be able to use the complete feature set of the Mumble protocol it is
+recommended that the client provides a strong certificate to the server.
+This however is not mandatory as you can connect to the server without
+providing a certificate. However the server must provide the client with
+its certificate and it is recommended that the client checks this.
+
+![Mumble connection setup](resources/mumble_connection_setup.png)
+
+## Version exchange
+
+Once the TLS handshake is completed both sides should transmit their version
+information using the Version message. The message structure is described below.
+
+| Field        | Type     |
+| ------------ | -------- |
+| `version`    | `uint32` |
+| `release`    | `string` |
+| `os`         | `string` |
+| `os_version` | `string` |
+
+The version field is a combination of major, minor and patch version numbers (e.g. 1.2.0)
+so that major number takes two bytes and minor and patch numbers take one byte each.
```

Comment:
Iirc this is now the legacy version format. We now use two bytes for every part.
I would tend to just mention the new format but would be fine with mentioning both.